### PR TITLE
DSRC-87: Hub Page fields

### PR DESCRIPTION
**Pull request by jamesbiggs**
Ticket URL: [DSRC-87]

## About these changes

Adds an `intro`, `hero_image` and `body` field to the `HubPage` model.

Also adds `links` which are a list of pages to be displayed as the hub pages links. These are either an internal or external page.

Added logic to prevent internal + external pages being added in the same card.

In the CMS:
image
image
image

In the API:
image
image

## Before assigning to reviewer, please make sure you have

- [x] Checked things thoroughly before handing over to reviewer
- [x] Checked PR title starts with ticket number as per project conventions to help us keep track of changes
- [x] Ensured that PR includes only commits relevant to the ticket
- [x] Waited for all CI jobs to pass before requesting a review
- [x] Added/updated tests and documentation where relevant

## Merging PR guidance

Follow [docs\developer-guide\contributing.md](https://nationalarchives.github.io/ds-wagtail/developer-guide/contributing/)

## Deployment guidance

Follow [docs\infra\environments.md](https://nationalarchives.github.io/ds-wagtail/infra/environments/)

[DSRC-87]: https://national-archives.atlassian.net/browse/DSRC-87?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by ahosgood** [comment]
@jamesbiggs For external links, could we set `full_url` (I know it will just be the same as `url`) so we have parity between the two? Also, could we have a `type_label` of "external"?

**Comment by ahosgood** [comment]
@jamesbiggs - Sorry, I meant could we have both `url` AND `full_url` for external links (even though they are the same).

That means that the frontend won't have to try and "decide" which one to use.

File: etna/generic_pages/models.py [hunk 1]
```diff
@@ -1,17 +1,28 @@
...
+    DefaultPageSerializer,
+    ImageSerializer,
+    RichTextSerializer,
+)
 
 from .blocks import GeneralPageStreamBlock
 
 
-class GeneralPage(BasePage):
+class GeneralPage(BasePageWithNonRequiredIntro):
```
Comment (ahosgood) [inline etna/generic_pages/models.py:25]: This feels...awkward 🤔

How difficult would it be to change the `BasePage` to not have a required introduction and then have the separate model for one that does require it?

Comment (ahosgood) [reply etna/generic_pages/models.py]: Okay, that sounds fair. I don't think I looked enough into the `BasePage` class.

Those two classes are a lot more readable than `BasePageWithNonRequiredIntro` 👍🏼

File: etna/core/models/basepage.py [hunk 2]
```diff
@@ -230,19 +232,7 @@ class BasePageWithNonRequiredIntro(BasePage):
...
-        null=True,
     )
 
     class Meta:
         abstract = True
-
-    content_panels = BasePage.content_panels + [FieldPanel("intro")]
-
-    search_fields = BasePage.search_fields + [
-        index.SearchField("intro", boost=3),
```
Comment (ahosgood) [inline etna/core/models/basepage.py]: Are we meant to have lost this?

Comment (jamesbiggs) [reply etna/core/models/basepage.py]: I have them on BasePageWithIntro and then BasePageWithRequiredIntro inherits BasePageWithIntro 😃

**Review by ahosgood — APPROVED** [review]

**Pull request merged**